src/com/seafile/seadroid2/cameraupload/CameraSyncAdapter.java

```diff
@@ -0,0 +1,686 @@
+
+        contentResolver = context.getContentResolver();
+        manager = new AccountManager(context);
+        dbHelper = CameraUploadDBHelper.getInstance();
+    }
+
+    private synchronized void startTransferService() {
+        if (txService != null)
+            return;
+
+        Intent bIntent = new Intent(getContext(), TransferService.class);
+        getContext().bindService(bIntent, mConnection, Context.BIND_AUTO_CREATE);
+    }
+
+    @Override
+    public void onSyncCanceled() {
+        super.onSyncCanceled();
+        Log.d(DEBUG_TAG, "onPerformSync will be cancelled ");
+        synchronized (this) {
+            cancelled = true;
+        }
+    }
+
+    private boolean isCancelled() {
+        synchronized (this) {
+            return cancelled;
+        }
+    }
+
+    /**
+     * Check if repository on the server exists.
+     *
+     * @param dataManager
+     * @return
+     * @throws SeafException
+     */
+    private boolean validateRepository(DataManager dataManager) throws SeafException {
+        List<SeafRepo> repos = dataManager.getReposFromServer();
+
+        for (SeafRepo repo : repos) {
+            if (repo.getID().equals(targetRepoId) && repo.getName().equals(targetRepoName))
+                return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Create all the subdirectories on the server for the buckets that are about to be uploaded.
+     *
+     * @param dataManager
+     * @throws SeafException
+     */
+    private void createDirectories(DataManager dataManager) throws SeafException {
+        List<GalleryBucketUtils.Bucket> buckets = GalleryBucketUtils.getMediaBuckets(getContext());
+
+        String serverPath = Utils.pathJoin(targetDir, BASE_DIR);
+
+        // recursively create directories in serverPath
+        createDirectoriesRecursively(dataManager, serverPath);
+
+        List<SeafDirent> dirs = dataManager.getDirentsFromServer(targetRepoId, serverPath);
+
+        for (GalleryBucketUtils.Bucket bucket : buckets) {
+
+            // the user has selected specific buckets: only create directories for these
+            if (!bucketList.isEmpty() && !bucketList.contains(bucket.id)) {
+                continue;
+            }
+
+            // auto-guessing is on: create directories for camera buckets
+            if (bucketList.isEmpty() && !bucket.isCameraBucket)
+                continue;
+
+            // first make sure that the bucket name exists and is a directory
+            boolean found = false;
+            for (SeafDirent dir : dirs) {
+                if (dir.name.equals(bucket.name) && dir.isDir()) {
+                    found = true;
+                } else if (dir.name.equals(bucket.name) && !dir.isDir()) {
+                    // there is already a file. move it away.
+                    dataManager.rename(targetRepoId,
+                            Utils.pathJoin(serverPath, dir.name),
+                            dir.name + " (renamed by Seadroid)",
```

Comment:
better to use `renamed by Seafile` and move it to strings.xml for the sake of i18n